app/src/main/java/net/accelf/devoptshide/HideDevOpts.kt

```diff
@@ -44,16 +52,19 @@ class HideDevOpts : IXposedHookLoadPackage {
         )
 
         private val callback = object : XC_MethodHook() {
-            override fun beforeHookedMethod(param: MethodHookParam?) {
+            override fun beforeHookedMethod(param: MethodHookParam) {
                 if (
-                    param == null
-                    || param.args[1] !is String
+                    param.args[1] !is String
                     || !names.contains(param.args[1] as String)
                 ) {
                     return
                 }
 
-                param.result = 0
+                if (param.method.getName() == "getInt") {
+                    param.result = 0
+                } else {
+                    param.result = "0"
+                }
```

Comment:
```suggestion
                param.result = when (param.method.getName()) {
                  "getInt" -> 0
                  "getString" -> "0"
                  else -> error("Illegal method name passed.")
                }
```

Reply:
Force-pushed to apply this (notice I used 4-space indentation rather, I hope it's fine with you).